### PR TITLE
Upgrade to next 13

### PR DESCRIPTION
**Pull request by Janpot**
Seems like core [wants to move fast](https://github.com/mui/material-ui/pull/35001) on this. So let's try to find the blockers for us to move to Next.js 13 as well.

- [x] Toolpad app OOM issues during build. (https://github.com/vercel/next.js/issues/40216 ?)
- [x] Toolpad app integration tests
  - [x] why are they about twice as slow now? middleware seems to have a big impact since 12.2, fixed in https://github.com/vercel/next.js/pull/42547. Confirmed on canary, 13.0.3 will contain the fix
  - [x] frameLocator is misbehaving in firefox. Tests become very flaky once middleware gets slow
  - [x] Failing duplication tests around custom components: https://github.com/mui/mui-toolpad/pull/1306
- [x] hydration mismatch when opening the docs
- [x] To do before merge:
  - [x] Upgrade `@mui/monorepo` dependency to master when https://github.com/mui/material-ui/pull/35001 is merged
  - [x] ~playwright 1.28~ not crucial for this PR
  - [x] next.13.0.3

### memory usage (`test_static`):
before (master):

Screenshot 2022-11-07 at 10 13 54

after:
Screenshot 2022-11-07 at 10 14 01

### CODE REVIEW

**Comment by mnajdova** [comment]
@Janpot should I wait for all issues here to be resolved before merging the PR on the core side?

**Comment by Janpot** [comment]
@mnajdova I'm decoupling the docs update from the toolpad application update in https://github.com/mui/mui-toolpad/pull/1297, that should allow you to move forward with https://github.com/mui/material-ui/pull/35001

**Comment by Janpot** [comment]
@mnajdova I got this green with only a few stable versions we're waiting on: `playwright@1.280` and `next@13.0.3`. No further blockers as far as I can see.

File: renovate.json [hunk 1]
```diff
@@ -69,11 +69,6 @@
       "matchPackageNames": ["core-js"],
       "allowedVersions": "< 2.0.0"
     },
-    {
-      "groupName": "next",
-      "matchPackageNames": ["next", "@next/eslint-plugin-next", "eslint-config-next"],
-      "allowedVersions": "1.2.2"
```
Comment (Janpot) [inline renovate.json:75]: 😄 Should've been `12.2.2`, but ok, the effect was the same

**Review by mnajdova — APPROVED** [review]

**Pull request merged**